server/sonar-server/src/main/java/org/sonar/server/sticker/ws/QualityGateAction.java

```diff
@@ -20,40 +20,117 @@
-  public static final String PARAM_TYPE = "type";
+  private static final String PARAM_COMPONENT = "component";
+  private static final String PARAM_BRANCH = "branch";
+  private static final String PARAM_TYPE = "type";
+
+  private static final String TYPE_BADGE = "BADGE";
+  private static final String TYPE_CARD = "CARD";
+
+  private static final String SVG_QUALITY_GATE_PASSING = "quality_gate-passing.svg";
+  private static final String SVG_QUALITY_GATE_WARNING = "quality_gate-warning.svg";
+  private static final String SVG_QUALITY_GATE_FAILING = "quality_gate-failing.svg";
+  private static final String SVG_NOT_FOUND = "not_found.svg";
+  private static final String SVG_UNAUTHORIZED = "unauthorized.svg";
+
+  private static final String STATUS_OK = "OK";
+  private static final String STATUS_WARNING = "WARN";
+  private static final String STATUS_ERROR = "ERROR";
+
+  private final UserSession userSession;
+  private final DbClient dbClient;
+
+  public QualityGateAction(UserSession userSession, DbClient dbClient) {
+    this.userSession = userSession;
+    this.dbClient = dbClient;
+  }
 
   @Override
   public void define(WebService.NewController controller) {
     NewAction action = controller.createAction("quality_gate")
       .setHandler(this)
-      .setDescription("Generate badge for quality gate as an SVG")
+      .setDescription("Generate badge for quality gate as an SVG.<br/>" +
+        "Requires the 'Browse' permission on the project.")
       .setResponseExample(Resources.getResource(getClass(), "quality_gate-example.svg"));
     action.createParam(PARAM_COMPONENT)
-      .setDescription("Project key")
+      .setDescription("Project key. When project does not exist, the badge will contain 'not found'.")
       .setRequired(true)
       .setExampleValue(KEY_PROJECT_EXAMPLE_001);
+    action
+      .createParam(PARAM_BRANCH)
+      .setDescription("Branch key")
+      .setExampleValue(KEY_BRANCH_EXAMPLE_001);
     action.createParam(PARAM_TYPE)
       .setDescription("Type of badge.")
-      .setRequired(false)
-      .setPossibleValues(asList("BADGE", "CARD"));
+      .setPossibleValues(asList(TYPE_BADGE, TYPE_CARD));
   }
 
   @Override
   public void handle(Request request, Response response) throws Exception {
     response.stream().setMediaType("image/svg+xml");
-    IOUtils.copy(Resources.getResource(getClass(), "quality_gate-example.svg").openStream(), response.stream().output());
+    try (DbSession dbSession = dbClient.openSession(false)) {
+      String projectKey = request.mandatoryParam(PARAM_COMPONENT);
+      String branch = request.param(PARAM_BRANCH);
+      Optional<ComponentDto> project = getProject(dbSession, projectKey, branch);
+      copy(Resources.getResource(getClass(), "templates/" + generateSvg(dbSession, project)).openStream(), response.stream().output());
+    }
+  }
+
+  private String generateSvg(DbSession dbSession, Optional<ComponentDto> projectOpt) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Specify a "ComponentDto" parameter instead. ([squid:S3553](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3553))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fjl%2FSONAR-10266%2Fquality_gate_badge_type&issues=AWDp9f4evOZGGKctVjX9&open=AWDp9f4evOZGGKctVjX9)

[ISSUE_KEY]: AWDp9f4evOZGGKctVjX9